scalameta/parsers/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala

```diff
@@ -2936,8 +2936,21 @@ class ScalametaParser(input: Input, dialect: Dialect) { parser =>
    *  }}}
    */
   def importees(): List[Importee] =
-    if (token.isNot[LeftBrace]) List(importWildcardOrName())
-    else inBraces(commaSeparated(importee()))
+    if (token.isNot[LeftBrace]) {
+      List(importWildcardOrName())
+    } else {
+      val importees = inBraces(commaSeparated(importee()))
+
+      if (importees.nonEmpty) {
+        importees.init.foreach { importee =>
+          if (importee.is[Importee.Wildcard]) {
+            syntaxError("Wildcard import must be in the last position", importee.pos)
+          }
+        }
+      }
```

Comment:
This would look a little nicer, but it would also needlessly allocate another list.